test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala

```diff
@@ -42,8 +42,10 @@ class EligibilityCheckConnectorSpec extends TestSupport with GeneratorDrivenProp
 
   implicit val resultArb: Arbitrary[EligibilityCheckResult] = Arbitrary(for {
     result ← Gen.alphaStr
+    resultCode ← Gen.choose(1, 10)
     reason ← Gen.alphaStr
-  } yield EligibilityCheckResult(result, reason))
+    reasonCode ← Gen.choose(1, 10)
+  } yield EligibilityCheckResult(result, resultCode, reason, reasonCode))
```

Comment:
the yield wont produce a valid `EligibilityCheckResult` entity here as its generating random strings and ints :)

Reply:
in the backend we don't particularly care if it's valid or not - the validation is done in the frontend. The backend is basically acting as a dumb proxy to DES for the frontend